src/qt/guiutil.h

```diff
@@ -239,6 +240,82 @@ namespace GUIUtil
     typedef ClickableProgressBar ProgressBar;
 #endif
 
+//! Wrapper class to serialize QDateTime objects as 32-bit time_t.
+template<typename Q>
+class AsTimeTWrapper
+{
+private:
+    Q& m_qdatetime;
+public:
+    AsTimeTWrapper(Q& qdatetime) : m_qdatetime(qdatetime) {}
+
+    template<typename Stream>
+    void Serialize(Stream& s) const { s << (uint32_t)m_qdatetime.toTime_t(); }
+
+    template<typename Stream>
+    void Unserialize(Stream& s)
+    {
+        uint32_t timeval;
+        s >> timeval;
+        m_qdatetime = QDateTime::fromTime_t(timeval);
+    }
+};
+template<typename Q> AsTimeTWrapper<Q> AsTimeT(Q& qdatetime) { return AsTimeTWrapper<Q>(qdatetime); }
+
+//! Wrapper class to serialize QString objects as std::strings.
+template<typename Q>
+class AsStdStringWrapper
+{
+private:
+    Q& m_qstring;
+public:
+    AsStdStringWrapper(Q& qstring) : m_qstring(qstring) {}
+
+    template<typename Stream>
+    void Serialize(Stream& s) const { s << m_qstring.toStdString(); }
+
+    template<typename Stream>
+    void Unserialize(Stream& s)
+    {
+        std::string str;
+        s >> str;
+        m_qstring = QString::fromStdString(std::move(str));
+    }
+};
+template<typename Q> AsStdStringWrapper<Q> AsStdString(Q& qstring) { return AsStdString<Q>(qstring); }
```

Comment:
This should say `return AsStdStringWrapper<Q>(qstring)` or just `return {qstring}`

Reply:
Fixed! Thanks for figuring this out!